### ISSUE TITLE
Optimize TensorFlow.js with WebML API with tf-js-convertor solution

### ISSUE THREAD

**Issue by BruceDai** [intel/webml-polyfill#11]

**Comment by BruceDai** [comment]
Decide to remove from open source milestone

**Comment by huningxin** [comment]
Postpone to next milestone.

**Comment by huningxin** [comment]
The steps of optimizing and converting TensorFlow MobileNet to TensorFlow.js format

1. Download Model
Model: http://download.tensorflow.org/models/mobilenet_v1_2018_02_22/mobilenet_v1_1.0_224.tgz
Refer to: https://github.com/tensorflow/models/tree/master/research/slim

2. Optimize Model
The original TensorFlow model contains `BatchNorm` layers which WebML API doesn't support yet. We use [toco](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/lite/toco) to fuse them into previous `Conv` layers.
```shell
$ toco --input_file=/path/to/mobilenet_v1_1.0_224/mobilenet_v1_1.0_224_frozen.pb --output_file=/path/to/optimized_graph.pb --input_format=TENSORFLOW_GRAPHDEF --output_format=TENSORFLOW_GRAPHDEF --input_shape=1,224,224,3 --input_array=input --output_array=MobilenetV1/Predictions/Reshape_1
```
3. Convert to TensorFlow.js format
Install [tensorflowjs_converter](https://github.com/tensorflow/tfjs-converter), then convert the model.

```shell
$ tensorflowjs_converter --input_format=tf_frozen_model --output_node_names='MobilenetV1/Predictions/Reshape_1' --saved_model_tags=serve /path/to/optimized_graph.pb /path/to/web_model/
```

**Comment by lisa0314** [comment]
Two solutions:
* graph executor (@huningxin ), quick&dirty `tf-js-convertor`
* Tensorflow kernel layer backend (@sunlin-link ), `tf-js-layers`

**Comment by huningxin** [comment]
Regarding to 

> graph executor (@huningxin ), quicktf-js-convertor

The prototype is hosted at: https://huningxin.github.io/tfjs-converter/demo/mobilenet_dist/

It executes mobilenet 100 times by tfjs graph executor with WebGL backend and WebML backend. Then it reports the average elapsed time and speedup.

The code has been pushed to https://github.com/huningxin/tfjs-converter/tree/webml and https://github.com/huningxin/tfjs-core/tree/webml

**Comment by halton** [comment]
Mark as done since tfjs-converter
